puppet/lib/puppet/util/puppetdb/char_encoding.rb

```diff
@@ -71,6 +71,20 @@ def self.utf8_string(str)
   def self.warn_if_changed(str, converted_str)
     if converted_str != str
       Puppet.warning "Ignoring invalid UTF-8 byte sequences in data to be sent to PuppetDB"
+
+      begin
+        fn = '/tmp/puppetdb-invalid-utf8-sequences'
+
+        analyser = EncodingAnalyser.new(fn)
+        analyser.original = str
+        analyser.converted = converted_str
+        analyser.write_summary
+
+        Puppet.warning "See #{fn} for details of last occurence"
+      rescue => e
```

Comment:
Is it necessary to do a blanket swallow of the exception here? Is it necessary to just swallow and return a warning, what is wrong about the default behaviour with exceptions within Puppet?